src/Markdown/Inline.elm

```diff
@@ -0,0 +1,98 @@
+module Markdown.Inline exposing
+    ( Inline(..)
+    , extractText
+    )
+
+{-| Inline rendering and helpers.
+
+
+# Model
+
+@docs Inline
+
+
+# Helpers
+
+@docs extractText
+
+-}
+
+import HtmlParser
+import Markdown.Block as Block
+
+
+{-| The inline type.
+
+  - **Text** | _Text_
+  - **HardLineBreak**
+  - **CodeInline** | _Code_
+  - **Link** | _Url_ | _Maybe Title_ | _Inlines_
+  - **Image** | _Source_ | _Maybe Title_ | _Inlines_
+  - **HtmlInline** | _Tag_ | _List ( Attribute, Maybe Value )_ | _Inlines_
+  - **Emphasis** | _Delimiter Length_ | _Inlines_
+
+-}
+type Inline htmlValue
+    = Text String
+    | HardLineBreak
+    | CodeInline String
+    | Link String (Maybe String) (List (Inline htmlValue))
+    | Image String (Maybe String) (List (Inline htmlValue))
+    | HtmlInline HtmlParser.Node
+      --| Html String (List Attribute) (List Block)
+    | Emphasis Int (List (Inline htmlValue))
```

Comment:
`htmlValue` is a phantom parameter here. Is that intentional?

Reply:
Indeed, good catch! Fixed 👍